SECURITY.md

```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
 
 
 ## Bug Bounty Program
-- as of May 2024, Vyper does not have a bug bounty program. It is planned to instantiate one soon.
+- Vyper runs a bug bounty program via Ethereum Foundation.
+  - Bugs should be reported through the [Ethereum Foundation's bounty program](https://bounty.ethereum.org/).
+
+### Scope
+- Rules from the Ethereum Foundation's bug bounty program apply. Here we further clarify the scope of the Vyper bounty program.
+- If a compiler bug affects production code, it is in scope.
+  - This includes bugs in older compiler versions still used in production (excluding known issues).
+- If a compiler bug does not currently affect production but is likely to in the future, it is in scope.
+  - This mainly applies to the latest compiler release (e.g., a new release is available but contracts are not yet deployed with it).
```

Comment:
there might be questions re what "new" exactly means (like 2 weeks, 6 months etc.). But we can address this if someone challenges it.

Reply:
Yeah, I agree, but it was intentional. I think it's realistic that 0.3.10 might be used, but how far into the past should we go? I leave it up to the hunters to convince us / ask further questions.

Reply:
I think `0.3.7` has been used widely but we shouldn't overthink it too much indeed.